plugins/heroku/src/herokuClient.ts

```diff
@@ -1,5 +1,30 @@
-import Heroku from 'heroku-client'
+import { ToolKitError } from '@dotcom-tool-kit/error'
+import { styles } from '@dotcom-tool-kit/logger'
+import { highlight } from 'cli-highlight'
+import Heroku, { HerokuError } from 'heroku-client'
 
 const HEROKU_AUTH_TOKEN = process.env.HEROKU_AUTH_TOKEN
 
-export default new Heroku({ token: HEROKU_AUTH_TOKEN })
+const client = new Heroku({ token: HEROKU_AUTH_TOKEN })
+
+// the Heroku client doesn't give particularly useful error messages, so we require an additional context
+// string to be included when calling the API which we include in custom error handling
+export async function herokuWithContext<T>(
+  context: string,
+  callback: (client: Heroku) => Promise<T>
+): Promise<T> {
+  try {
+    return await callback(client)
+  } catch (e) {
+    const upstream = e as HerokuError
+    const error = new ToolKitError('there was an error calling the Heroku API')
+    error.details = `received the error ${styles.errorHighlight(upstream.message)} when ${context}`
+    if (upstream.body) {
+      error.details += `\n\nresponse from Heroku was:\n${highlight(
+        JSON.stringify(upstream.body, undefined, 2),
```

Comment:
**suggestion**: Heroku errors always have `id` and `message` fields, and sometimes a `url`, can we just print the `message` and `url`? think it would be cleaner than outputting JSON. the `id` seems to be just [based on the status code](https://devcenter.heroku.com/articles/platform-api-reference#client-error-responses), so we can skip that.